common/src/main/java/org/opensearch/sdk/DeleteDataObjectResponse.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+package org.opensearch.sdk;
+
+import org.opensearch.core.common.Strings;
+import org.opensearch.core.index.shard.ShardId;
+
+public class DeleteDataObjectResponse {
+    private final String id;
+    private final ShardId shardId;
```

Comment:
Is `shardId` required? This can tie implementation specific to opensearch?

Reply:
Unfortunately, yes. I didn't add anything I didn't have to.  

This was required for `DeleteConnectorTransportAction` which processes an `ActionListener<DeleteResponse>` (which is eventually parsed and returned to the user) as a response to the delete API:
```json
{
    "_index": "the_index",
    "_id": "vZAyLY8BaPsQZP-UDULE",
    "_version": 2,
    "result": "deleted",
    "_shards": {
        "total": 1,
        "successful": 1,
        "failed": 0
    },
    "_seq_no": 1,
    "_primary_term": 1
}
```
It's not allowed to be null:
```java
this.shardId = Objects.requireNonNull(shardId);
```
I can manufacture one (and the builder provides the facility to do that) but it's gotta be there. 😦